2021_CommandBased/src/main/java/frc/robot/subsystems/Drive_s.java

```diff
@@ -38,6 +41,10 @@
 
     //create odometry
     XdriveOdometry odometry;
+
+    private final XdrivePoseEstimator poseEstimator;
+
+    private AHRS imu;
```

Comment:
I would prefer if shared resources like motor controllers and IMUs were in the RobotContainer class.

Reply:
with command based robots, the convention is to put hardware in subsystems and subsystems in RobotContainer

Reply:
After looking at the drive subsystem a bit more, I don't think there would be a great need to unit test since the logic is mostly in other objects.